programs/discard_server.c

```diff
@@ -105,7 +105,7 @@ receive_cb(struct socket *sock, union sctp_sockstore addr, void *data,
 			       rcv.rcv_sid,
 			       rcv.rcv_ssn,
 			       rcv.rcv_tsn,
-			       ntohl(rcv.rcv_ppid),
```

Comment:
Why are you casting the result? `ntohl()` and alike return an unsigned integer: [Posix](https://pubs.opengroup.org/onlinepubs/009695399/functions/htonl.html). This comment applies also to other similar changes.

Reply:
To fix a compiler warning for MinGW with GCC, clang is fine with the code without casting it.
`htonl()` on Windows returns an `u_long` [docs here](https://docs.microsoft.com/en-us/windows/desktop/api/winsock/nf-winsock-htonl)

`u_long` gets translated to `unsigned long`

Since it doesn't harm the any other platform, I didn't write another workaround.

```
[ 49%] Building C object programs/CMakeFiles/discard_server_upcall.dir/discard_server_upcall.c.obj
C:/Documents and Settings/weinrank/Documents/GitHub/usrsctp/programs/discard_server_upcall.c: In function 'handle_upcall':
C:/Documents and Settings/weinrank/Documents/GitHub/usrsctp/programs/discard_server_upcall.c:124:93: error: format '%u' expects argument of type 'unsigned int', but argument 8 has type 'u_long' {aka 'long unsigned int'} [-Werror=format=]
  124 |     printf("Msg of length %d received from %s:%u on stream %d with SSN %u and TSN %u, PPID %u, context %u.\n",
      |                                                                                            ~^
      |                                                                                             |
      |                                                                                             unsigned int
      |                                                                                            %lu
......
  131 |            ntohl(rn.recvv_rcvinfo.rcv_ppid),
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |            |
      |            u_long {aka long unsigned int}
cc1.exe: all warnings being treated as errors
make[2]: *** [programs/CMakeFiles/discard_server_upcall.dir/build.make:63: programs/CMakeFiles/discard_server_upcall.dir/discard_server_upcall.c.obj] Error 1
make[1]: *** [CMakeFiles/Makefile2:301: programs/CMakeFiles/discard_server_upcall.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

Reply:
I don't see a problem here. Windows uses `unsigned long`, which is the same as `unsigned int`, since Windows is a LLP64 system. Both are 32-bit unsigned integers. System other than Windows often use ILP32, which means the `unsigned int` is a 32-bit integer and `unsigned long` is a 64-bit entity.

So I guess this is a bug in gcc. Can we leave the code untouched, since it is correct, and just turn off the warning when using gcc on Windows?